### PR TITLE
Scma/spring data

### PR DESCRIPTION
**Pull request by RoadRunner14**

### CODE REVIEW

File: camp/src/main/java/ch/bfh/zodnrk/camp/Application.java [hunk 1]
```diff
@@ -10,10 +11,11 @@ public static void main(String[] args) {
         final AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(ServiceConfig.class);
 
         final PartyService partyService = context.getBean(PartyService.class);
+        final HeroService heroService = context.getBean(HeroService.class);
         Party fellowship = partyService.createParty("The Fellowship");
 
         System.out.printf("Created party '%s' with %d members.%n", fellowship.getName(), fellowship.getMembers().size());
-
+        heroService.printStrongHeros(50);
```
Comment (martyschaer) [inline camp/src/main/java/ch/bfh/zodnrk/camp/Application.java]: Small typo: 
```suggestion
        heroService.printStrongHeroes(50);
```

File: camp/src/main/java/ch/bfh/zodnrk/camp/model/Hero.java [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
+    @Id
+    @GeneratedValue(generator = "system-uuid")
+    @GenericGenerator(name = "system-uuid", strategy = "uuid")
+    private String id;
     private String name;
     private int atk;
     private int def;
     private double hp;
 
+    public void setId(String id) {
```
Comment (martyschaer) [inline camp/src/main/java/ch/bfh/zodnrk/camp/model/Hero.java]: Order is usually getter, then setter.

File: camp/src/main/java/ch/bfh/zodnrk/camp/service/HeroService.java [hunk 3]
```diff
@@ -4,4 +4,6 @@
 
 public interface HeroService {
     Hero createHero(String name);
+
+    void printStrongHeros(int atkLimit);
```
Comment (martyschaer) [inline camp/src/main/java/ch/bfh/zodnrk/camp/service/HeroService.java]: Small typo: 
```suggestion
    void printStrongHeroes(int atkLimit);
```

**Review by martyschaer — CHANGES_REQUESTED** [review]
Some small fixes remain, but generally looks good!

File: camp/src/main/java/ch/bfh/zodnrk/camp/model/Hero.java [hunk 4]
```diff
@@ -1,12 +1,31 @@
...
     private String name;
     private int atk;
     private int def;
     private double hp;
 
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
```
Comment (cyrillbolliger) [inline camp/src/main/java/ch/bfh/zodnrk/camp/model/Hero.java]: I suggest removing the setter for a UUID field, we should not mess around with UUIDs manually

File: camp/src/main/java/ch/bfh/zodnrk/camp/service/impl/DefaultHeroService.java [hunk 5]
```diff
@@ -1,17 +1,20 @@
...
 import java.util.Random;
 
 @Named
 public class DefaultHeroService implements HeroService {
 
-    private final Random rng = new Random();
-
     private static final double DEFAULT_HERO_HEALTH = 100.0;
+    private final Random rng = new Random();
+    @Autowired
```
Comment (cyrillbolliger) [inline camp/src/main/java/ch/bfh/zodnrk/camp/service/impl/DefaultHeroService.java]: We should not mix Spring & JSR330 annotations:

- Spring: `@Service` & `@Autowired`
- JSR330: `@Named` & `@Inject`

I suggest going with the JSR330 standard

**Review by cyrillbolliger — APPROVED** [review]
:1st_place_medal:

File: camp/src/main/java/ch/bfh/zodnrk/camp/service/impl/DefaultHeroService.java (hunk 5 above)
Comment (martyschaer) [reply camp/src/main/java/ch/bfh/zodnrk/camp/service/impl/DefaultHeroService.java]: That's a good point.
We started with using [constructor Injection](https://github.com/zodnrk/project-heroes/blob/master/camp/src/main/java/ch/bfh/zodnrk/camp/service/conf/ServiceConfig.java#L18) for better testability.
We should either continue that, or switch to field injection everywhere.

I also like the JSR330 standard better.

Comment (cyrillbolliger) [reply camp/src/main/java/ch/bfh/zodnrk/camp/service/impl/DefaultHeroService.java]: Ah, yeah, your totally right. I'd suggest we continue with constructor injection. Thanks for pointing this out.

**Pull request merged**